File: backend/entity/notification.go
package entity

import (
	"gorm.io/gorm"
)

type Notification struct {
	gorm.Model
	Title   string `json:"title"`
	Type    string `json:"type"`    // e.g. "payment", "system", ...
	Message string `json:"message"`
	UserID  uint   `json:"user_id"`
	User    *User  `gorm:"foreignKey:UserID" json:"user"`
}

File: frontend/src/interfaces/Notification.ts
// src/interfaces/Notification.ts
import type { User } from "./User";

export interface Notification {
  ID: number;
  title: string;
  type: string;   // e.g. "payment", "system"
  message: string;
  user_id: number;
  user?: User;
}

export interface CreateNotificationRequest {
  title: string;
  type: string;
  message: string;
  user_id: number;
}

export interface UpdateNotificationRequest {
  ID: number;
  title?: string;
  type?: string;
  message?: string;
}
